# Showing Yankees analysis especially focusing on A. Benintendi trade

In [24]:
from pybaseball import batting_stats, team_game_logs, statcast_batter, playerid_lookup
import numpy as np
import pandas as pd

In [2]:
data = batting_stats(2022, qual = 200)
data

IDfg  Season               Name   Team  Age   G   AB   PA    H  1B  ...  \
2    15640    2022        Aaron Judge    NYY   30  95  361  417  107  53  ...   
1     9218    2022   Paul Goldschmidt    STL   34  93  352  405  118  66  ...   
13    9777    2022      Nolan Arenado    STL   31  91  345  384  102  59  ...   
8    13510    2022       Jose Ramirez    CLE   29  92  349  398  101  46  ...   
34   18314    2022     Dansby Swanson    ATL   28  99  381  416  112  75  ...   
..     ...     ...                ...    ...  ...  ..  ...  ...  ...  ..  ...   
251  27465    2022  Spencer Torkelson    DET   22  83  264  298   52  36  ...   
260  13723    2022    Jacob Stallings    MIA   32  76  221  242   43  37  ...   
264  19948    2022     Cristian Pache    OAK   23  68  201  214   32  24  ...   
256  10071    2022    Jonathan Villar  - - -   31  59  202  220   42  31  ...   
210  11737    2022   Nick Castellanos    PHI   30  96  374  401   92  64  ...   

     Barrel%  maxEV  HardHit  HardHit%  Events  CStr%   CSW%    xBA   xSLG  \
2      0.258  118.4      156     0.600     260  0.183  0.295  0.296  0.683   
1      0.122  112.3      121     0.448     270  0.195  0.293  0.263  0.492   
13     0.080  111.4      114     0.381     299  0.166  0.258  0.258  0.423   
8      0.061  111.6      118     0.376     314  0.156  0.209  0.258  0.409   
34     0.118  109.0      125     0.460     272  0.149  0.289  0.265  0.484   
..       ...    ...      ...       ...     ...    ...    ...    ...    ...   
251    0.068  111.5       72     0.379     190  0.173  0.279  0.218  0.342   
260    0.024  104.8       48     0.293     164  0.217  0.314  0.203  0.282   
264    0.041  108.6       64     0.435     147  0.156  0.298  0.228  0.305   
256    0.027  109.5       58     0.397     146  0.114  0.268  0.208  0.283   
210    0.072  110.1       95     0.342     278  0.108  0.282  0.246  0.391   

     xwOBA  
2    0.440  
1    0.368  
13   0.327  
8    0.326  
34   0.351  
..     ...  
251  0.291  
260  0.249  
264  0.256  
256  0.247  
210  0.300  

[265 rows x 319 columns]

In [3]:
Judge = data.query('Name == "Aaron Judge" ')['HR']
Bad_hit = data.query('H <= 38 ')[['Name', 'Team', 'G', 'PA', 'AB', 'H', 'AVG','HR', 'OPS']].sort_values(by = 'PA', ascending = False)
Bad_hit

Name Team   G   PA   AB   H    AVG  HR    OPS
217      Joey Gallo  NYY  80  270  230  37  0.161  12  0.629
264  Cristian Pache  OAK  68  214  201  32  0.159   2  0.427
262  Brett Phillips  TBR  72  202  178  26  0.146   5  0.480

In [4]:
Team_result = team_game_logs(2022, "NYY")[['Rslt', 'HR']]
Team_result['Y_HR'] = np.where(Team_result['HR'] > 0, 1,0)
Team_result['Result'] = np.where(Team_result['Rslt'].str[0] == 'W', 1,0)
Num_games = Team_result.shape[0]

In [5]:
Team_result

Rslt  HR  Y_HR  Result
0   W,6-5   3     1       1
1   W,4-2   2     1       1
2   L,3-4   0     0       0
3   L,0-3   0     0       0
4   W,4-0   1     1       1
..    ...  ..   ...     ...
93  L,5-7   2     1       0
94  W,7-6   3     1       1
95  L,3-6   1     1       0
96  W,6-0   1     1       1
97  L,3-6   2     1       0

[98 rows x 4 columns]

In [6]:
Bi_Hr = Team_result[['Y_HR', 'Result']]
Bi = Bi_Hr.value_counts().reset_index()
Bi

Y_HR  Result   0
0     1       1  56
1     1       0  19
2     0       0  13
3     0       1  10

In [7]:
Bi[0]/Num_games

0    0.571429
1    0.193878
2    0.132653
3    0.102041
Name: 0, dtype: float64

In [8]:
56/(56+19)

0.7466666666666667

## 60% of wins NYY hit a HR
## if a HR is hit 75% chance of winning

In [10]:
Judge_log = pd.read_excel(r"C:\Users\Owner\OneDrive\Desktop\exploratory data files\Judge Game Log.xlsx")
Judge_log1 = Judge_log
Judge_log1

Rk   Gcar  Gtm                 Date   Tm Unnamed: 5  Opp   Rslt   Inngs  \
0    1.0  573.0    1  2022-04-08 00:00:00  NYY        NaN  BOS  W,6-5  CG(11)   
1    2.0  574.0    2  2022-04-09 00:00:00  NYY        NaN  BOS  W,4-2      CG   
2    3.0  575.0    3  2022-04-10 00:00:00  NYY        NaN  BOS  L,3-4      CG   
3    4.0  576.0    4  2022-04-11 00:00:00  NYY        NaN  TOR  L,0-3      CG   
4    5.0  577.0    5  2022-04-12 00:00:00  NYY        NaN  TOR  W,4-0      CG   
..   ...    ...  ...                  ...  ...        ...  ...    ...     ...   
91  92.0  664.0   95  2022-07-22 00:00:00  NYY          @  BAL  W,7-6      CG   
92  93.0  665.0   96  2022-07-23 00:00:00  NYY          @  BAL  L,3-6      CG   
93  94.0  666.0   97  2022-07-24 00:00:00  NYY          @  BAL  W,6-0      CG   
94  95.0  667.0   98  2022-07-26 00:00:00  NYY          @  NYM  L,3-6      CG   
95   NaN    NaN  NaN                  NaN  NaN        NaN  NaN  63-32     NaN   

     PA  ...    OPS  BOP   aLI    WPA  acLI    cWPA   RE24  DFS(DK)  DFS(FD)  \
0     5  ...  1.000  2.0  1.21 -0.002  1.18  0.0000  -0.22       10     12.2   
1     4  ...  0.708  2.0  1.01  0.004  1.04  0.0000  -0.16        4      6.2   
2     5  ...  0.742  3.0  1.86 -0.054  1.93 -0.0003   0.36       11     12.0   
3     4  ...  0.646  3.0  1.28 -0.044  1.34 -0.0003  -0.32        2      3.0   
4     4  ...  0.668  2.0  0.56 -0.037  0.58 -0.0002   0.46        5      6.0   
..  ...  ...    ...  ...   ...    ...   ...     ...    ...      ...      ...   
91    4  ...  1.003  2.0  0.98  0.327  0.30  0.0006   2.97       32     44.4   
92    5  ...  1.016  2.0  1.18  0.267  0.34  0.0005   3.01       25     31.0   
93    5  ...  1.025  2.0  0.47  0.147  0.15  0.0003   2.02       23     31.4   
94    5  ...  1.035  2.0  1.44  0.100  0.35  0.0001   1.11       19     24.7   
95  417  ...  1.035  NaN  1.07  3.107  0.88  0.0180  32.74     1090   1461.2   

      Pos  
0      RF  
1   CF RF  
2      RF  
3      RF  
4   CF RF  
..    ...  
91     CF  
92     CF  
93  DH CF  
94     CF  
95    NaN  

[96 rows x 40 columns]

In [11]:
Judge_log1['Result'] = np.where(Judge_log1['Rslt'].str[0] == 'W', 1,0)
Judge_log1['Y_HR'] = np.where(Judge_log1['HR'] > 0, 1,0)
Judge_log1['Y_H'] = np.where(Judge_log1['H'] > 0, 1,0)
Judge_log1['MH'] = np.where(Judge_log1['H'] > 1, 1,0)
Judge_log1 = Judge_log1.head(Judge_log1.shape[0] - 1)
Judge_log1

Rk   Gcar Gtm                 Date   Tm Unnamed: 5  Opp   Rslt   Inngs  \
0    1.0  573.0   1  2022-04-08 00:00:00  NYY        NaN  BOS  W,6-5  CG(11)   
1    2.0  574.0   2  2022-04-09 00:00:00  NYY        NaN  BOS  W,4-2      CG   
2    3.0  575.0   3  2022-04-10 00:00:00  NYY        NaN  BOS  L,3-4      CG   
3    4.0  576.0   4  2022-04-11 00:00:00  NYY        NaN  TOR  L,0-3      CG   
4    5.0  577.0   5  2022-04-12 00:00:00  NYY        NaN  TOR  W,4-0      CG   
..   ...    ...  ..                  ...  ...        ...  ...    ...     ...   
90  91.0  663.0  94           Jul 21 (2)  NYY          @  HOU  L,5-7      CG   
91  92.0  664.0  95  2022-07-22 00:00:00  NYY          @  BAL  W,7-6      CG   
92  93.0  665.0  96  2022-07-23 00:00:00  NYY          @  BAL  L,3-6      CG   
93  94.0  666.0  97  2022-07-24 00:00:00  NYY          @  BAL  W,6-0      CG   
94  95.0  667.0  98  2022-07-26 00:00:00  NYY          @  NYM  L,3-6      CG   

    PA  ...  acLI    cWPA  RE24  DFS(DK)  DFS(FD)    Pos  Result  Y_HR  Y_H  \
0    5  ...  1.18  0.0000 -0.22       10     12.2     RF       1     0    1   
1    4  ...  1.04  0.0000 -0.16        4      6.2  CF RF       1     0    0   
2    5  ...  1.93 -0.0003  0.36       11     12.0     RF       0     0    1   
3    4  ...  1.34 -0.0003 -0.32        2      3.0     RF       0     0    0   
4    4  ...  0.58 -0.0002  0.46        5      6.0  CF RF       1     0    1   
..  ..  ...   ...     ...   ...      ...      ...    ...     ...   ...  ...   
90   5  ...  0.19 -0.0001  1.32       22     31.9     DH       0     1    1   
91   4  ...  0.30  0.0006  2.97       32     44.4     CF       1     1    1   
92   5  ...  0.34  0.0005  3.01       25     31.0     CF       0     0    1   
93   5  ...  0.15  0.0003  2.02       23     31.4  DH CF       1     1    1   
94   5  ...  0.35  0.0001  1.11       19     24.7     CF       0     1    1   

    MH  
0    1  
1    0  
2    1  
3    0  
4    0  
..  ..  
90   0  
91   1  
92   1  
93   1  
94   1  

[95 rows x 44 columns]

In [17]:
df_Jud = Judge_log1[['Y_HR', 'Result']]
df_Jud.value_counts()

Y_HR  Result
0     1         40
      0         25
1     1         23
      0          7
dtype: int64

In [16]:
23/30

0.7666666666666667

## if a HR is hit by judge 77% chance of winning

In [15]:
df_Jud1 = Judge_log1[['Y_H', 'Result']]
df_Jud1.value_counts()

Y_H  Result
1    1         47
     0         18
0    1         16
     0         14
dtype: int64

In [14]:
47/95

0.49473684210526314

In [18]:
47/(47+18)

0.7230769230769231

## in games we win about 50% of them judge gets a hit
## in games where he gets a hit we win 72% of the time

In [19]:
df_Jud2 = Judge_log1[['MH', 'Result']]
df_Jud2.value_counts()

MH  Result
0   1         39
    0         24
1   1         24
    0          8
dtype: int64

In [20]:
24/95

0.25263157894736843

In [21]:
24/32

0.75

## about 25% of our wins judge had a multihit game
## 75% of the multihit games we won

In [93]:
playerid_lookup("stanton")

name_last name_first  key_mlbam key_retro  key_bbref  key_fangraphs  \
0   stanton       buck     122676  stanb101  stantbu01        1012418   
1   stanton  giancarlo     519317  stanm004  stantmi03           4949   
2   stanton      leroy     122678  stanl102  stantle01        1012420   
3   stanton       mike     122681  stanm003  stantmi02            849   
4   stanton       mike     122679  stanm001  stantmi01        1012421   
5   stanton        tom     122680  stant101  stantto01        1012422   

   mlb_played_first  mlb_played_last  
0            1931.0           1931.0  
1            2010.0           2022.0  
2            1970.0           1978.0  
3            1989.0           2007.0  
4            1975.0           1985.0  
5            1904.0           1904.0

In [94]:
data1 = statcast_batter('2022-04-01', '2022-07-28', player_id = 592450)
data2 = statcast_batter('2022-04-01', '2022-07-28', player_id = 519203)
data3 = statcast_batter('2022-04-01', '2022-07-28', player_id = 650402)
data4 = statcast_batter('2022-04-01', '2022-07-28', player_id = 519317)

Gathering Player Data
Gathering Player Data
Gathering Player Data
Gathering Player Data


In [79]:
test = data1[['player_name','events', 'on_1b', 'on_2b', 'on_3b']]
HR_Judge = test.query('(events == "home_run") | (events == "single") | (events == "double")')
HR_Judge['no_runners'] = np.where((HR_Judge['on_1b'].isna()) & (HR_Judge['on_2b'].isna()) & (HR_Judge['on_3b'].isna()) , 1,0)
HR_Judge[['player_name', 'events', 'no_runners']].value_counts()

<ipython-input-79-e74f5aa2a136>:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  HR_Judge['no_runners'] = np.where((HR_Judge['on_1b'].isna()) & (HR_Judge['on_2b'].isna()) & (HR_Judge['on_3b'].isna()) , 1,0)


player_name   events    no_runners
Judge, Aaron  single    0             29
                        1             27
              home_run  1             26
                        0             14
              double    0              9
                        1              8
dtype: int64

In [88]:
test1 = data2[['player_name','events', 'on_1b', 'on_2b', 'on_3b']]
HR_Rizzo = test1.query('(events == "home_run") | (events == "single") | (events == "double")')
HR_Rizzo['no_runners'] = np.where((HR_Rizzo['on_1b'].isna()) & (HR_Rizzo['on_2b'].isna()) & (HR_Rizzo['on_3b'].isna()) , 1,0)
HR_Rizzo[['player_name', 'events', 'no_runners']].value_counts()

<ipython-input-88-1423d10ff872>:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  HR_Rizzo['no_runners'] = np.where((HR_Rizzo['on_1b'].isna()) & (HR_Rizzo['on_2b'].isna()) & (HR_Rizzo['on_3b'].isna()) , 1,0)


player_name     events    no_runners
Rizzo, Anthony  single    0             20
                          1             17
                home_run  1             15
                double    0              9
                home_run  0              9
                double    1              7
dtype: int64

In [89]:
test2 = data3[['player_name','events', 'on_1b', 'on_2b', 'on_3b']]
HR_torres = test2.query('(events == "home_run") | (events == "single") | (events == "double")')
HR_torres['no_runners'] = np.where((HR_torres['on_1b'].isna()) & (HR_torres['on_2b'].isna()) & (HR_torres['on_3b'].isna()) , 1,0)
HR_torres[['player_name', 'events', 'no_runners']].value_counts()

<ipython-input-89-e1e8d8fa3186>:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  HR_torres['no_runners'] = np.where((HR_torres['on_1b'].isna()) & (HR_torres['on_2b'].isna()) & (HR_torres['on_3b'].isna()) , 1,0)


player_name      events    no_runners
Torres, Gleyber  single    1             26
                           0             22
                 home_run  1             11
                 double    1             10
                           0              8
                 home_run  0              6
dtype: int64

In [95]:
test3 = data4[['player_name','events', 'on_1b', 'on_2b', 'on_3b']]
HR_stanton = test3.query('(events == "home_run") | (events == "single") | (events == "double")')
HR_stanton['no_runners'] = np.where((HR_stanton['on_1b'].isna()) & (HR_stanton['on_2b'].isna()) & (HR_stanton['on_3b'].isna()) , 1,0)
HR_stanton[['player_name', 'events', 'no_runners']].value_counts()

<ipython-input-95-ac018b68f41c>:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  HR_stanton['no_runners'] = np.where((HR_stanton['on_1b'].isna()) & (HR_stanton['on_2b'].isna()) & (HR_stanton['on_3b'].isna()) , 1,0)


player_name         events    no_runners
Stanton, Giancarlo  single    1             22
                              0             15
                    home_run  0             13
                              1             11
                    double    0              4
                              1              2
dtype: int64

# When Aaron Judge performs well it generally results in a Yankee win:
### 75% of multi-hit games result in a win
### 73% of games where he gets a hit, yankees win
### 77% win rate when he hits a HR

## Having Benintendi at the beg of the lineup will help because there is high% of HR being hit by middle of the lineup players with no runners on
### And about 50% of singles and doubles as well. Putting a .300 batting average player at the beginning of the lineup will help